pkg/controller/route/routecontroller.go

```diff
@@ -50,6 +50,11 @@ func New(routes cloudprovider.Routes, kubeClient clientset.Interface, clusterNam
 }
 
 func (rc *RouteController) Run(syncPeriod time.Duration) {
+	// TODO: If we do just the full Resync every 5 minutes (default value)
+	// that means that we may wait up to 5 minutes before even starting
+	// creating a route for it. This is bad.
+	// We should have a watch on node and if we observe a new node (with CIDR?)
+	// trigger reconciliation for that node.
```

Comment:
Yes - this is a long standing issue (it's nothing new). I will send a PR for it, but that won't be trivial change, so I'm not sure if it will land in 1.3.
But yeah - #26267 would makes nodes to be not ready in such case.